PhysicsTools/Heppy/python/analyzers/core/TriggerBitAnalyzer.py

```diff
@@ -12,12 +12,18 @@ def __init__(self, cfg_ana, cfg_comp, looperName ):
         self.fallbackName = getattr(self.cfg_ana,"fallbackProcessName",None)
         self.outprefix   = getattr(self.cfg_ana,"outprefix",  self.processName)
         self.unrollbits = ( hasattr(self.cfg_ana,"unrollbits") and self.cfg_ana.unrollbits )
- 
+        self.saveIsUnprescaled = getattr(self.cfg_ana,"saveIsUnprescaled",False)
+        self.force1prescale = False
+        if self.cfg_comp.isMC and self.saveIsUnprescaled:
+            print 'Cannot save prescale information in MC: will put everything to unprescaled'
+            self.saveIsUnprescaled = False
+            self.force1prescale = True
 
     def declareHandles(self):
         super(TriggerBitAnalyzer, self).declareHandles()
         fallback = ('TriggerResults','',self.fallbackName) if self.fallbackName else None
-        self.handles['TriggerResults'] = AutoHandle( ('TriggerResults','',self.processName), 'edm::TriggerResults', fallbackLabel=fallback )
+        self.handles['TriggerResults'] = AutoHandle( ('TriggerResults','',self.processName), 'edm::TriggerResults', fallbackLabel=(('TriggerResults','',self.fallbackName) if self.fallbackName else None) )
+        if self.saveIsUnprescaled: self.handles["TriggerPrescales"] = AutoHandle( ('patTrigger','',self.processName), 'pat::PackedTriggerPrescales', fallbackLabel=(('patTrigger','',self.fallbackName) if self.fallbackName else None) )
```

Comment:
For MiniAOD version 2 we should also check the l1 prescales (see WorkBookMiniAOD2015)